jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java

```diff
@@ -59,12 +61,14 @@ public RemoteReporter(final Sender sender, int flushInterval, int maxQueueSize,
     commandQueue = new ArrayBlockingQueue<>(maxQueueSize);
 
     // start a thread to append spans
+    long serialNumber = nextSerialNumber.getAndIncrement();
```

Comment:
I am not sure why we need this. In a real application there would be only one thread with each name.